lib/ruby_lsp/tapioca/addon.rb

```diff
@@ -23,24 +25,33 @@ def initialize
 
         @global_state = T.let(nil, T.nilable(RubyLsp::GlobalState))
         @rails_runner_client = T.let(nil, T.nilable(RubyLsp::Rails::RunnerClient))
+        @index = T.let(nil, T.nilable(RubyIndexer::Index))
+        @file_checksums = T.let({}, T::Hash[String, String])
+        @outgoing_queue = T.let(nil, T.nilable(Thread::Queue))
       end
 
       sig { override.params(global_state: RubyLsp::GlobalState, outgoing_queue: Thread::Queue).void }
       def activate(global_state, outgoing_queue)
         @global_state = global_state
         return unless @global_state.enabled_feature?(:tapiocaAddon)
 
+        @index = @global_state.index
+        @outgoing_queue = outgoing_queue
         Thread.new do
           # Get a handle to the Rails add-on's runtime client. The call to `rails_runner_client` will block this thread
           # until the server has finished booting, but it will not block the main LSP. This has to happen inside of a
           # thread
           addon = T.cast(::RubyLsp::Addon.get("Ruby LSP Rails", ">= 0.3.17", "< 0.4"), ::RubyLsp::Rails::Addon)
           @rails_runner_client = addon.rails_runner_client
-          outgoing_queue << Notification.window_log_message("Activating Tapioca add-on v#{version}")
+          @outgoing_queue << Notification.window_log_message("Activating Tapioca add-on v#{version}")
           @rails_runner_client.register_server_addon(File.expand_path("server_addon.rb", __dir__))
         rescue IncompatibleApiError
           # The requested version for the Rails add-on no longer matches. We need to upgrade and fix the breaking
           # changes
+          outgoing_queue << Notification.window_log_message(
```

Comment:
Should this also be an instance variable?

```suggestion
          @outgoing_queue << Notification.window_log_message(
```